test/TestExtensions/NuGet.StaFact/GlobalSuppressions.cs

```diff
@@ -0,0 +1,10 @@
+// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "VSTHRD002:Avoid problematic synchronous waits", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.StaFact.NuGetWpfTestCase.CopyTaskResultFrom``1(System.Threading.Tasks.TaskCompletionSource{``0},System.Threading.Tasks.Task{``0})")]
+[assembly: SuppressMessage("Usage", "VSTHRD003:Avoid awaiting foreign Tasks", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.StaFact.NuGetWpfTestCase.RunAsync(Xunit.Abstractions.IMessageSink,Xunit.Sdk.IMessageBus,System.Object[],Xunit.Sdk.ExceptionAggregator,System.Threading.CancellationTokenSource)~System.Threading.Tasks.Task{Xunit.Sdk.RunSummary}")]
+[assembly: SuppressMessage("Usage", "VSTHRD110:Observe result of async calls", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.StaFact.NuGetWpfTestCase.RunAsync(Xunit.Abstractions.IMessageSink,Xunit.Sdk.IMessageBus,System.Object[],Xunit.Sdk.ExceptionAggregator,System.Threading.CancellationTokenSource)~System.Threading.Tasks.Task{Xunit.Sdk.RunSummary}")]
```

Comment:
These should have real justifications, rather than "pending"

Reply:
The VS threading library is a library created by the VS team, rather than a library only useful in VS. Its analyzers could potentially be useful for any app with a UI thread and I've seen AArnott recomment the vs threading library to people creating standalone WPF apps. `STAFact` appears to exist precisely to emulate a UI thread in a unit test, although the tests using `STAFact` would benefit from the analyzer, rather than `STAFact` itself.

The VSTHRD002 can possibly be silenced by putting everything after the `if (!template.IsComplete) { throw .. }` block in an else block, so that the analyzer knows it's only possible to reach when the task is already complete.

The other two cases, I'm not sure what to suggest, because these method level suppressions doesn't give me enough information to know which line in the method are triggering the analyzer.

Having said that, I'm surprised the PackageReference with ExcludeAssets didn't disable the analyzer. I'm not sure what else to suggest. I think VS has some GUI options to enable/disable/change severity of analyzers, so you could try that, check how it made the changes, and then make any adjustments necessary to ensure it only applies to the STAFact project and not to the whole solution.

Reply:
Added justifications for 002 and 003. Eliminated need for 110 using same patterns a FileAndForget.